src/power_grid_model/validation/validation.py

```diff
@@ -331,7 +379,25 @@ def validate_required_values(
         required["line"] += ["r0", "x0", "c0", "tan0"]
         required["shunt"] += ["g0", "b0"]
 
-    return list(chain(*(none_missing(data, component, required.get(component, [])) for component in data)))
+    # Mark `power_sigma` for each power_sensor based on the state of `p_sigma` and `q_sigma`
+    process_power_sigma_and_p_q_sigma(data, "sym_power_sensor", required)
+    process_power_sigma_and_p_q_sigma(data, "asym_power_sensor", required)
+
+    return list(
+        chain(
+            *(
+                none_missing(data, component, item, index)
+                for component in data
+                for index, item in enumerate(
+                    (sublist for sublist in required.get(component, []))
+                    if isinstance(required.get(component, []), list)
+                    and all(isinstance(i, list) for i in required.get(component, []))
+                    else [required[component]]
+                )
+                if data[component] is not None and isinstance(data[component], Sized) and index < len(data[component])
```

Comment:
this statement is so complex that it for me suggests that this is probably not the right solution. maybe earlier you should've just created a `np.array` instead of a `list` to enable this?

IMO, the original return statement was still readable but now it's gotten a bit too much for me.

while i am very much for chaining in the modern programming sense (or similarly in the Lisp sense), not everything is more readable in one statement. remember the fork bomb? `:(){:|:&};:`

Reply:
Well, when it comes to this sort of topic, it is always a matter of degree. 
While 
> not the right solution

is clearly not the right wording, I agree with you that it might be too much for one statement. However, For its purpose of tailoring the `required` field according to `data`, it might not need further modification.